error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/CanonicalConstantFieldName.java

```diff
@@ -0,0 +1,184 @@
+   *
+   * @param flags Any provided command line flags.
+   */
+  @Inject
+  CanonicalConstantFieldName(ErrorProneFlags flags) {
+    optionalExcludedConstantFliedNames = getCanonicalizedLoggerName(flags);
+    includePublicConstantFieldNames = isIncludePrivateConstantFieldNames(flags);
+  }
+
+  @Override
+  public Description matchCompilationUnit(CompilationUnitTree tree, VisitorState state) {
+    ImmutableList.Builder<VariableTree> variablesInCompilationUnitBuilder = ImmutableList.builder();
+    new TreeScanner<@Nullable Void, @Nullable Void>() {
+      @Override
+      public @Nullable Void visitClass(ClassTree classTree, @Nullable Void unused) {
+        for (Tree member : classTree.getMembers()) {
+          if (member.getKind() == Kind.VARIABLE) {
+            variablesInCompilationUnitBuilder.add((VariableTree) member);
+          }
+        }
+        return super.visitClass(classTree, unused);
+      }
+    }.scan(tree, null);
+
+    ImmutableList<VariableTree> variables = variablesInCompilationUnitBuilder.build();
+    if (variables.isEmpty()) {
+      return Description.NO_MATCH;
+    }
+
+    ImmutableList<VarSymbol> variableSymbols =
+        variables.stream().map(ASTHelpers::getSymbol).collect(toImmutableList());
+    SuggestedFix.Builder fixBuilder = SuggestedFix.builder();
+    variables.forEach(
+        variableTree -> {
+          if (IS_CONSTANT.matches(variableTree, state)
+              && isFieldAccessModifierApplicable(variableTree, state)) {
+            VarSymbol variableSymbol = ASTHelpers.getSymbol(variableTree);
+            String variableName = variableSymbol.getSimpleName().toString();
+
+            if (!isVariableUpperSnakeCase(variableName) && !isVariableNameExcluded(variableName)) {
+              String replacement = toUpperSnakeCase(variableName);
+
+              if (variableSymbols.stream()
+                  .noneMatch(s -> s.getSimpleName().toString().equals(replacement))) {
+                fixBuilder.merge(SuggestedFixes.renameVariable(variableTree, replacement, state));
+              } else {
+                reportConstantRenameBlocker(variableTree, replacement, state);
+              }
+            }
+          }
+        });
+
+    return fixBuilder.isEmpty() ? Description.NO_MATCH : describeMatch(tree, fixBuilder.build());
+  }
+
+  private void reportConstantRenameBlocker(
+      VariableTree tree, String replacement, VisitorState state) {
+    state.reportMatch(
+        buildDescription(tree)
+            .setMessage(
+                String.format(
+                    "Suggested fix for constant name conflicts with an already defined variable `%s`.",
+                    replacement))
+            .build());
+  }
+
+  private boolean isFieldAccessModifierApplicable(VariableTree tree, VisitorState state) {
+    return includePublicConstantFieldNames || IS_PRIVATE.matches(tree, state);
+  }
+
+  private static boolean isVariableUpperSnakeCase(String variableName) {
```

Comment:
The method name does not have something to do with `variable`, so we can omit that part :).